pkg/datagatherer/k8s/dynamic_test.go

```diff
@@ -237,11 +242,25 @@ func TestDynamicGatherer_Fetch(t *testing.T) {
 			if err == nil && test.err {
 				t.Errorf("expected to get an error but didn't get one")
 			}
-			if diff, equal := messagediff.PrettyDiff(test.expected, res); !equal {
-				t.Errorf("\n%s", diff)
-				expectedJSON, _ := json.MarshalIndent(test.expected, "", "  ")
-				gotJSON, _ := json.MarshalIndent(res, "", "  ")
-				t.Fatalf("unexpected JSON: \ngot \n%s\nwant\n%s", string(gotJSON), expectedJSON)
+
+			if test.expected != nil {
+				list, ok := res.(*unstructured.UnstructuredList)
+				if !ok {
+					t.Errorf("expected result be an UnstructuredList but wasn't")
+				}
+
+				// TODO this is pointless, we can make the expected value an unstructured.UnstructuredList and avoid this check
```

Comment:
Might be worth just using the list as the expected value rather than asserting the type from an interace{} we set up in this very test.